server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,21 +363,63 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
+            // TODO: Add support for parsing the otel incoming tracer.
+            final SpanScope httpRequestSpanScope = tracer.startSpan(createSpanName(httpRequest), buildSpanAttributes(httpRequest));
+            httpRequestSpanScope.addSpanAttribute(AttributeNames.HTTP_REQ_INBOUND_EX, httpRequest.getInboundException() != null);
```

Comment:
Why is this attribute for `HTTP_REQ_INBOUND_EX`? I think I already commented on that: when the error is dispatched, we should end up span with `error(httpRequest.getInboundException())`, because this is what we would send to the client.

Reply:
This was at 2 places I guess. Removing.